config/admin/services.yml

```diff
@@ -28,3 +28,9 @@ services:
   PrestaShop\Module\APIResources\Validation\IframeValidationGroupsResolver:
     arguments:
       - '@prestashop.adapter.legacy.configuration'
+
+  # Custom denormalizer to handle type conversion for SearchProducts query
+  PrestaShop\Module\APIResources\Normalizer\SearchProductsDenormalizer:
```

Comment:
The denormalizer approach is interesting and handles the problem smartly But I'm afraid this will have to be replicated too many times, I'm surprized there is nothing to handle this in the API Platform Filters system https://api-platform.com/docs/core/filters/